lib/mail/version_specific/ruby_1_9.rb

```diff
@@ -125,7 +125,11 @@ def Ruby19.q_value_decode(str)
     def Ruby19.param_decode(str, encoding)
       str = uri_parser.unescape(str)
       str = charset_encoder.encode(str, encoding) if encoding
-      str
+      decoded = str.encode(Encoding::UTF_8, :invalid => :replace, :replace => "")
+      decoded.valid_encoding? ? decoded : decoded.encode(Encoding::UTF_16LE, :invalid => :replace, :replace => "").encode(Encoding::UTF_8)
+    rescue Encoding::UndefinedConversionError, ArgumentError, Encoding::ConverterNotFoundError
```

Comment:
I copied the strategy used in the two other _decode methods with another PR coming to refactor this into a shared method that all three will use.

Should I make that other PR first?